DAO.sol

```diff
@@ -615,6 +626,47 @@ contract DAO is DAOInterface, Token, TokenCreation {
         p.open = false;
     }
 
+    function withdraw() noEther onlyTokenholders returns (bool _success) {
+
+        // don't allow withdrawing when token holder is blocked due to a vote
+        // unVoteAll() would be an alternative
+        if (isSplitBlocked(msg.sender))
+            throw;
+
+        // Move ether
+        uint fundsToBeMoved =
+            (balances[msg.sender] * actualBalance()) /
+            totalSupply;
+
+        msg.sender.call.value(fundsToBeMoved);
```

Comment:
**DANGER:** Here we are gonna become vulnerable to the recursive call exploit again! 

1. Change it to `send()`
2. Make all changes to the "state of the contract" before the actual send/call to protect from any recursive calls.